test/Microsoft.Azure.WebJobs.Host.FunctionalTests/HostListenerFactoryTests.cs

```diff
@@ -37,36 +38,70 @@ public HostListenerFactoryTests()
             DisableProvider_Instance.Method = null;
         }
 
-        [Fact]
-        public async Task CreateAsync_RegistersScaleMonitors()
+        [Theory]
+        [InlineData(false, typeof(TestListener_Monitor), typeof(TestListener_Monitor), 2, 0)]
+        [InlineData(false, typeof(TestListener_Monitor), typeof(TestListener_TargetScaler), 1, 0)]
+        [InlineData(false, typeof(TestListener_Monitor), typeof(TestListener_TargetScalerProvider), 1, 0)]
+        [InlineData(false, typeof(TestListener_TargetScaler), typeof(TestListener_TargetScalerProvider), 0, 0)]
+        [InlineData(false, typeof(TestListener_Monitor), typeof(TestListener_MonitorAndTargetScaler), 2, 0)]
+        [InlineData(true, typeof(TestListener_Monitor), typeof(TestListener_Monitor), 2, 0)]
+        [InlineData(true, typeof(TestListener_Monitor), typeof(TestListener_TargetScaler), 1, 1)]
+        [InlineData(true, typeof(TestListener_Monitor), typeof(TestListener_TargetScalerProvider), 1, 1)]
+        [InlineData(true, typeof(TestListener_TargetScaler), typeof(TestListener_TargetScalerProvider), 0, 2)]
+        [InlineData(true, typeof(TestListener_Monitor), typeof(TestListener_MonitorAndTargetScaler), 1, 1)]
+        public async Task CreateAsync_RegistersScaleMonitorsAndTargetScaler(bool tbsEnabled,
```

Comment:
nit: CreateAsync_RegistersScaleMonitorsAndTargetScalers

Reply:
The test is not needed anymore as we are registering everything and function host decides what to use